jsonrpc/dispatcher.go

```diff
@@ -330,13 +337,15 @@ func (d *Dispatcher) handleReq(req Request) ([]byte, error) {
 	}
 	respBytes, err := json.Marshal(resp)
 	if err != nil {
-		return nil, d.internalError(req.Method, err)
+		return nil, d.internalError(req.ID, req.Method, err)
 	}
 	return respBytes, nil
 }
 
-func (d *Dispatcher) internalError(method string, err error) error {
+func (d *Dispatcher) internalError(id interface{}, method string, err error) error {
 	d.logger.Error("failed to dispatch", "method", method, "err", err)
+	internalError.Data = id
```

Comment:
I think we do this method a couple of times here in the JSON-RPC Layer: We have a global error object that we modify before sending. Why wouldn't we create a new one each time, instead of modifying the same one, and by that risking some data to be leftover?

We can use the same principle from txpool: Have constant error messages, but create new errors each time